lib/IRGen/GenClangDecl.cpp

```diff
@@ -54,6 +59,29 @@ class ClangDeclFinder
     return true;
   }
 
+  bool VisitFunctionDecl(clang::FunctionDecl *functionDecl) {
+    for (auto paramDecl : functionDecl->parameters()) {
+      if (paramDecl->hasDefaultArg()) {
+        if (FuncDecl *defaultArgGenerator =
+                clangModuleLoader->getDefaultArgGenerator(paramDecl)) {
+          // Deconstruct the Swift function that was created in
+          // SwiftDeclSynthesizer::makeDefaultArgument and extract the
+          // underlying Clang function that was also synthesized.
+          BraceStmt *body = defaultArgGenerator->getTypecheckedBody();
+          auto returnStmt =
+              cast<ReturnStmt>(body->getSingleActiveElement().get<Stmt *>());
+          auto callExpr = cast<CallExpr>(returnStmt->getResult());
+          auto calledFuncDecl = cast<FuncDecl>(callExpr->getCalledValue());
+          auto calledClangFuncDecl =
+              cast<clang::FunctionDecl>(calledFuncDecl->getClangDecl());
+          callback(calledClangFuncDecl);
+        }
+      }
+    }
```

Comment:
So if we ever reference a function, we have to eagerly emit default argument generators for all of its parameters?  Is there some more reasonable way to make SIL carry the information for how to generate the body here so that we can just trigger this when we have a reference to the default argument generator?

Reply:
We don't actually emit the generator unless it is used. Clang's CodeGen adds this decl to the list of deferred decls (https://github.com/apple/llvm-project/blob/d4738f0f52791c6004131af7322f13a07abee8e7/clang/lib/CodeGen/CodeGenModule.cpp#L3679C2-L3679C2) and emits it only if it's referenced.

I added an IRGen test to make sure unused generators aren't emitted.

Reply:
Ah, that makes sense.